Implementação dos métodos Patch e Delete + Remoção da propriedade Ida…

ocaiooo opened the pull request:
…de pela adição da propriedade Data Nascimento

ocaiooo committed:
Implementação dos métodos Patch e Delete + Remoção da propriedade Idade pela adição da propriedade Data Nascimento

tiags38 commented on .idea/.idea.TimesBD/.idea/.gitignore:
remover essa pasta e colocar ela no gitignore

tiags38 commented on TimesBD/Controllers/TimesController.cs:
```diff
@@ -40,9 +44,28 @@ public async Task<IActionResult> Get([FromQuery(Name = "name")]string name = nul
+
+    [HttpPatch("{id}")]
+    public async Task<IActionResult> Patch(int id, JogadorModel atualizaJogador)
+    {
```
fazer validações afim de verificar se os campos descritos são validos e se correspondem ao que se deve. 
-verificar os tipos se correspondem 
-defina uma data de nascimento maxima e minima. Exemplo, nascimento n pode ser no futuro 
-altera o campo Time_id no banco para TimeId

tiags38 commented on TimesBD/Controllers/TimesController.cs:
```diff
@@ -63,7 +86,7 @@ public async Task<IActionResult> Post(Jogador jogador)
                 jogador.Siafi = endereco.Siafi;
                 
-                await sqlConnection.ExecuteAsync("INSERT INTO Jogadores (Nome, Idade, Time_id, Cep, Logradouro, Complemento, Bairro, Localidade, Uf, Ibge, Gia, Ddd, Siafi) VALUES (@Nome, @Idade, @Time_id, @Cep, @Logradouro, @Complemento, @Bairro, @Localidade, @Uf, @Ibge, @Gia, @Ddd, @Siafi)", jogador);
+                await sqlConnection.ExecuteAsync("INSERT INTO Jogadores (Nome, DataNascimento, Time_id, Cep, Logradouro, Complemento, Bairro, Localidade, Uf, Ibge, Gia, Ddd, Siafi) VALUES (@Nome, @DataNascimento , @Time_id, @Cep, @Logradouro, @Complemento, @Bairro, @Localidade, @Uf, @Ibge, @Gia, @Ddd, @Siafi)", jogador);
```
fazer mesmas validações que tem no update

tiags38 commented on TimesBD/Controllers/TimesController.cs:
```diff
@@ -72,6 +95,22 @@ public async Task<IActionResult> Post(Jogador jogador)
+    {
+        using (var sqlConnection = new SqlConnection(_connectionString))
+        {
+            var linhaAfetada = await sqlConnection.ExecuteAsync("DELETE FROM Jogadores WHERE Id = @id", new { id });
```
-criar esse Id como parametro 
?id=1 ao inves de /1

tiags38 commented on TimesBD/Controllers/TimesController.cs:
```diff
@@ -72,6 +95,22 @@ public async Task<IActionResult> Post(Jogador jogador)
+
+            if (linhaAfetada == 0)
+            {
+                return NotFound();
```
coloca um texto no retorno, informando que o Id nao foi encontrado

tiags38 commented on TimesBD/TimesBD.csproj:
```diff
@@ -21,4 +21,5 @@
     </ItemGroup>
 
 
+
```
remover todas as linahs em branco

tiags38 commented on TimesBD/Models/JogadorModel.cs:
```diff
@@ -0,0 +1,14 @@
+    public DateTime DataNascimento { get; set; }
+    
+    [JsonProperty("time_id")]
+    public int Time_id { get; set; }
```
alterar para TimeId

tiags38 reviewed (changes requested):
Fazer as alterações solicitadas

ocaiooo committed:
Update .gitignore

ocaiooo committed:
Delete .idea/.idea.TimesBD/.idea/.gitignore

ocaiooo committed:
Update TimesBD.csproj retirando linhas em branco

ocaiooo committed:
Alterações solicitadas no último pull request

ocaiooo committed:
Merge branch 'dev' of https://github.com/CaioCeesaar/TimesBD into dev

ocaiooo committed:
Verificação se nome é igual a "string"

Fiz uma atualização para que se o nome for igual a "string" a gente continua com o nome antigo
